<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>11</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Task1</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTreeWidget" name="treeWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>301</width>
      <height>440</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string>Tree</string>
     </property>
    </column>
   </widget>
   <widget class="QTreeWidget" name="treeWidget_2">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>10</y>
      <width>230</width>
      <height>330</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string notr="true">Subtree</string>
     </property>
    </column>
   </widget>
   <widget class="QSpinBox" name="spinKey1">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>30</y>
      <width>111</width>
      <height>30</height>
     </rect>
    </property>
    <property name="maximum">
     <number>1000000000</number>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineValue1">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>110</y>
      <width>113</width>
      <height>30</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="mainTreeKeyLabel">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>10</y>
      <width>110</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Key</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>70</y>
      <width>110</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Value</string>
    </property>
   </widget>
   <widget class="QPushButton" name="addButton1">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>30</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Add</string>
    </property>
   </widget>
   <widget class="QPushButton" name="removeButton1">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>70</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Remove(key)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="findButton1">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>110</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Find(key)</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinKey2">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>370</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="maximum">
     <number>1000000000</number>
    </property>
   </widget>
   <widget class="QPushButton" name="addButton2">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>370</y>
      <width>110</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Add branch</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>350</y>
      <width>66</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Key</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>400</y>
      <width>66</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Value</string>
    </property>
   </widget>
   <widget class="QPushButton" name="removeButton2">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>420</y>
      <width>110</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Remove by key</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineValue2">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>420</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="lowestCommonAncestorButton">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>370</y>
      <width>230</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Find lowest common ancestor</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin1">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>420</y>
      <width>111</width>
      <height>30</height>
     </rect>
    </property>
    <property name="maximum">
     <number>1000000000</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>400</y>
      <width>100</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Key1</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin2">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>420</y>
      <width>111</width>
      <height>30</height>
     </rect>
    </property>
    <property name="maximum">
     <number>1000000000</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>400</y>
      <width>110</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Key2</string>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>151</y>
      <width>230</width>
      <height>190</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QPushButton" name="insertSubtreeButton">
       <property name="text">
        <string>Insert subtree</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="removeSubtreeButton">
       <property name="text">
        <string>Remove subtree</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="insertLBranchButton">
       <property name="text">
        <string>Insert left branch</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="insertRBranchButton">
       <property name="text">
        <string>Insert right branch</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="removeLBranchButton">
       <property name="text">
        <string>Remove left branch</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="removeRBranchButton">
       <property name="text">
        <string>Remove right branch</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
